File: apps/crud-be-java-vertx/src/test/java/com/demobejavx/unit/ConfigSecurityTest.java
package com.demobejavx.unit;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static org.junit.jupiter.api.Assertions.assertFalse;

/**
 * RED: MainVerticle.java uses getOrDefault(..., DEFAULT_JWT_SECRET) so missing var falls back silently.
 * GREEN will replace with an explicit null-check that throws on missing CRUD_BE_JAVA_VERTX_JWT_SECRET.
 */
class ConfigSecurityTest {

    @Test
    void mainVerticleHasNoSoftDefaultForJwtSecret() throws IOException {
        String src = Files.readString(
            Path.of("src/main/java/com/demobejavx/MainVerticle.java")
        );
        assertFalse(
            src.contains("getOrDefault(\"CRUD_BE_JAVA_VERTX_JWT_SECRET\""),
            "MainVerticle must not use getOrDefault for JWT_SECRET — "
                + "absent var must throw, not silently fall back"
        );
    }
}

File: apps/crud-be-kotlin-ktor/src/test/kotlin/com/demobektkt/unit/ConfigSecurityTest.kt
package com.demobektkt.unit

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test
import java.nio.file.Files
import java.nio.file.Path

/**
 * RED: Application.kt uses `?: "dev-jwt-secret-..."` so missing var falls back silently.
 * GREEN will replace with `error(...)` / `requireNotNull(...)` so absent var terminates startup.
 */
class ConfigSecurityTest {

    @Test
    fun `Application_kt has no soft default for jwt secret`() {
        val src = Files.readString(
            Path.of("src/main/kotlin/com/demobektkt/Application.kt")
        )
        assertFalse(
            src.contains("?: \"dev-jwt-secret"),
            "Application.kt must not use a soft default for JWT_SECRET — " +
                "absent var must call error() or requireNotNull(), not fall back"
        )
    }
}
